Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -157,7 +131,12 @@ void Serializer::EncodeTypedValue(const TypedValue& value,
         // TODO(rsgowman): figure out error handling
         abort();
       }
-      EncodeBool(&stream, value.value.boolean_value);
+      if (field_value == FieldValue::TrueValue()) {
```

Comment:
Why not `field_value.boolean_value()`?

Reply:
Because it doesn't exist. (Though it should.)

I want to keep it out of this PR (as this is essentially a refactoring) but will add it to a followup: PR #862

Reply:
SG.

Please wait for the M22 freeze to lift before merging. Alternatively, if enough of these are stacking up let's target a post-m22 branch.